test/tests/functional/pbs_resc_custom_perm.py

```diff
@@ -0,0 +1,164 @@
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class Test_Custom_Resource_Perm(TestFunctional):
+    rsc_list = ['foo_str', 'foo_strm', 'foo_strh']
+    hook_list = ["start", "begin"]
+
+    def tearDown(self):
+        self.du.set_pbs_config(confs={'PBS_SERVER': self.server.hostname})
+        TestFunctional.tearDown(self)
+        for rsc in self.rsc_list:
+            try:
+                self.server.manager(MGR_CMD_DELETE, RSC,
+                                    id=rsc, runas=ROOT_USER)
+            except PbsManagerError:
+                pass
+        for hook in self.hook_list:
+            self.server.manager(MGR_CMD_DELETE, HOOK, id=hook, runas=ROOT_USER)
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        for rsc in self.rsc_list:
+            try:
+                self.server.manager(MGR_CMD_DELETE, RSC,
+                                    id=rsc, runas=ROOT_USER)
+            except PbsManagerError:
+                pass
+
+        self.momA = self.moms.values()[0]
+        self.momA.restart()
+        self.momA.delete_vnode_defs()
+        self.hostA = self.momA.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+        self.server.expect(NODE, {'state': 'free'}, id=self.hostA)
+
+        # Next set custom resources with m flag and without m flag
+        # via qmgr -c 'create resource'
+        attr = {'type': 'string', 'flag': 'h'}
+        r = 'foo_str'
+        rc = self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, runas=ROOT_USER, logerr=False)
+        self.assertEqual(rc, 0)
+
+        attr = {'type': 'string', 'flag': 'hm'}
+        r = 'foo_strm'
+        rc = self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, runas=ROOT_USER, logerr=False)
+        self.assertEqual(rc, 0)
+
+        # Create a custom resources via exechost_startup hook.
+        startup_hook_body = """
+import pbs
+e=pbs.event()
+localnode=pbs.get_local_nodename()
+e.vnode_list[localnode].resources_available['foo_strh'] = "str_hook"
+"""
+        hook_name = "start"
+        a = {'event': "exechost_startup", 'enabled': 'True'}
+        rv = self.server.create_import_hook(
+            hook_name,
+            a,
+            startup_hook_body,
+            overwrite=True)
+        self.assertTrue(rv)
+
+        self.momA.restart()
```

Comment:
Why are you restarting the mom?  The hook will be sent to the mom after it is created.

Reply:
Well, this restart is needed to run a startup hook on the MoM which creates a custom resource.

Reply:
Ahh, got it.  I didn't think about that.  Thanks for the explanation.